src/Template/Variable.php

```diff
@@ -15,10 +15,13 @@ final class Variable implements NameTypeItem, JsonSerializable
 
     private Type $type;
 
-    public function __construct(string $name, Type $type)
+    private bool $mightBeUndefined;
+
+    public function __construct(string $name, Type $type, bool $mightBeUndefined = false)
     {
         $this->name = $name;
         $this->type = $type;
+        $this->mightBeUndefined = $mightBeUndefined;
```

Comment:
Once it might be use when variable is assigned in condition in presenter like:
```php
if ($someCondition) {
    $this->template->a = 'abcd';
}
```
We just have to resolve if variable is assigned in all branches of the method or not, but may be there is some logic in phpstan itself, we shouldn't reinvent the wheel.

Reply:
Not sure if it is possible to solve. We encountered similar problem before with terminating methods and did not find a solution. In my projects I solved it by using temp variable and assigning to template outside condition.

Reply:
Note: It would also need to cover cases when variable is assigned in conditionaly called method/function.

Reply:
I said it will be possible mar variable as optionally defined :) I just don't know how to achieve it :))